### PR TITLE
Separates headers and data in BedRock networks

### PR DESCRIPTION
**Pull request by dpetrisko**
## Summary
This PR separates out the headers and data in the bedrock networks. This has several benefits. First, it reduces the need to have separate declarations for identical headers, and removes doubts that they can be cast to each other. Second, it moves things closer to deprecating bp lite entirely. The current implementation in a lot of modules are basically lite == stream - last, which implies that it is an interface which does not support burst. Of course, the ultimate goal is to convert all interface to stream or burst as necessary. But for e.g. the LCEs, this goal may be a ways off. 

## Area
ME

## Additional Changes Required (if any)
bsg_fpga and zynq-parrot will likely need to change slightly. Because there's no functional impact, I'm not tying together PRs in those downstream repos to this one.

## Analysis
No PPA impact, pure refactor.

## Verification
Full regression.

### CODE REVIEW

**Comment by dpetrisko** [comment]
(disable whitespace changes for a better diff)

File: .gitlab-ci.yml
Comment (dpetrisko) [inline .gitlab-ci.yml:29]: will remove before merge

File: bp_me/src/v/lce/bp_lce_cmd.sv
Comment (muwyse) [inline bp_me/src/v/lce/bp_lce_cmd.sv:317]: delete this line

File: bp_me/src/v/lce/bp_lce_cmd.sv
Comment (muwyse) [inline bp_me/src/v/lce/bp_lce_cmd.sv:359]: delete line

File: bp_me/src/v/lce/bp_lce_cmd.sv
Comment (muwyse) [inline bp_me/src/v/lce/bp_lce_cmd.sv:550]: delete line

File: bp_me/src/v/lce/bp_lce_cmd.sv
Comment (muwyse) [inline bp_me/src/v/lce/bp_lce_cmd.sv:667]: delete line

**Review by muwyse — APPROVED** [review]
LGTM!

Delete a few extra lines then good to go if passing regressions.

File: bp_me/src/v/lce/bp_lce_cmd.sv
Comment (muwyse) [reply bp_me/src/v/lce/bp_lce_cmd.sv:317]: This is remnants of needing to assign the payload struct into the payload field of the header. The BedRock interface macros now properly instantiate the payload struct in the header so we assign directly to it using hdr.payload.*

**Pull request merged**
